### PR TITLE
feat(Network): implement request.redirectChain() api

### LINKED ISSUE #1816: Expose redirect chain for requests
The `request.redirectChain()` should return all the requests that are a part of the redirect chain for the `request`.

```js
const response = await page.goto('http://google.com');
const requests = response.request().redirectChain();
for (const request of requests)
  console.log(request.url());
```

This was requested a few times, e.g. https://github.com/GoogleChrome/puppeteer/issues/1708.

### PR DESCRIPTION
**Pull request by aslushnikov**
The request.redirectChain() returns all the requests issued to fetch
the resource.

Fixes #1816.

### CODE REVIEW

**Comment by JoelEinbinder** [comment]
`Response.fromServiceWorker` failed on cirrus. Is that related to this patch?

File: lib/NetworkManager.js [hunk 1]
```diff
@@ -377,6 +384,13 @@ class Request {
     return this._frame;
   }
 
+  /**
+   * @return {!Array<!Request>}
+   */
+  redirectChain() {
+    return this._redirectChain;
```
Comment (JoelEinbinder) [inline lib/NetworkManager.js]: It looks like we are returning a live array that might change after the user receives it. Slice here?

**Comment by aslushnikov** [comment]
@JoelEinbinder i don't think so. I think I fix flakiness in https://github.com/GoogleChrome/puppeteer/pull/2195 (btw, ptal)

**Comment by aslushnikov** [comment]
@JoelEinbinder addressed comments; ptal.

**Review by JoelEinbinder — APPROVED** [review]

**Pull request merged**